Add bearer authorization to gradle wrapper (#32102)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/reference/runtime-configuration/gradle_wrapper.adoc

[[sec:authenticated_download]]
=== Authenticated Gradle distribution download

> The Gradle `Wrapper` can download Gradle distributions from servers using HTTP Basic Authentication or with a static HTTP Bearer Token.

Reviewer: There is a section below stating that Basic Authentication should only be used with HTTPS. That's just as important for bearer tokens too, so that section should be updated too.

This can be used in conjunction with a proxy, authenticated or not.
See <<networking.adoc#sec:accessing_the_web_via_a_proxy,Accessing the web via a proxy>> for more information on how to configure the `Wrapper` to use a proxy.

> WARNING: If you use `systemProp.gradle.wrapperUser` and `systemProp.gradle.wrapperPassword` to download the Gradle Wrapper from a private server, your credentials may be leaked if a build attempts to download the wrapper from a different server..

Reviewer 2: ```suggestion
WARNING: Using `systemProp.gradle.wrapperUser` and `systemProp.gradle.wrapperPassword` may leak your credentials if the build attempts to download the Gradle Wrapper from a server other than your private one.
```

To safeguard your credentials, prefix the system properties with the private server's hostname, replacing all dots (`.`) with underscores (`_`).

For example, if your private server hostname is `your.private-server.com`, use the following system properties:

[source,properties]
----
systemProp.gradle.your_private-server_com.wrapperUser=username
systemProp.gradle.your_private-server_com.wrapperPassword=password
----

To authenticate with a static HTTP Bearer Token, add the following line to your system properties file:

[source,properties]
----
systemProp.gradle.your_private-server_com.wrapperToken=some-api-token
----

NOTE: If a `wrapperToken` is specified, it will take precedence over `wrapperUser` and `wrapperPassword`.

> It is strongly recommended to include the hostname in the property to prevent the token from being accidentally sent to an unintended server.

Reviewer 2: ```suggestion
It is strongly recommended to include the hostname in the property to prevent the token from being sent to an unintended server by accident.
```
